jekyll/_posts/2018-06-09-discover-godot-nim.md

```diff
@@ -0,0 +1,93 @@
+Thinking in nodes is the first thing I had to learn. Everything in Godot is a node and a tree of nodes make up a scene. Your game could be one scene or multiple scenes, it depends on how you tackle the design of your game. Personally I find it a lot easier to make a project in Godot if I make a rough sketch on paper of all the important nodes in a game and how they work together.
+
+### Godot + Nim = ???
+
+Somewhere halfway I stopped using GDScript and started my journey on figuring out how the hell the [Nim bindings](https://github.com/pragmagic/godot-nim) work. This... wasn't the smartest of choices as there aren't many examples on how to work with the bindings. The [godotnim-stub](https://github.com/pragmagic/godot-nim-stub) is an excellent starting place if you have previous experience, but as a beginner I was continously searching for hints, even looking through other GDNative bindings trying to find a way to do something which is so simple in GDScript. Looking back I definitely would be more productive in GDScript and I'd understand Godot alot better.
+_But!_ I did finish 2 small samples following the ["Your first game"](http://docs.godotengine.org/en/3.0/getting_started/step_by_step/your_first_game.html) tutorial and a [Conways Game of Life](http://www.brainjargames.com/game-of-life/) post using Nim instead of GDScript. Using types made things a lot clearer in code and I learned the awesome feeling of tackling a problem that might seem impossible to figure out at first, but to which the answer is so simple once solved.
+For those who are also keen to try Nim with Godot, I'd say take a look at the stub project and the api the bindings generate. My experience following the GDScript focused tutorials with Nim was that it all translates pretty well. Nim compiles so fast that I didn't find myself growing tired of the compile script -> press f5 in godot cycle.
+
+```nim
+import godot
+import animation_player, control
+
+type
+  CellKind* {.pure.} = enum # the asterisk means it's exported
+    Living
+    Dead
+
+gdobj Cell of Control:
+  var kind: CellKind
+  var x*, y*: int
+  var anim: AnimationPlayer
+
+  method ready*() =
+    anim = getNode("AnimationPlayer").as(AnimationPlayer)
+
+  proc initTile*(x,y: int) =
+    self.x = x
+    self.y = y
+
+  proc setKind*(kind: CellKind) =
+    self.kind = kind
+    case kind:
+    of CellKind.Living:
+      anim.play("Living")
+    of CellKind.Dead:
+      anim.play("Dead")
+
+  proc getKind*(): CellKind =
+    kind
+```
+
+##### A lot of thanks to @endragor, one of the maintainers of the plugin for answering my often stupid questions
+
+`gdobj` is the macro goodness that makes extending a node so easy. Declaring fields, properties and methods all happen inside it. Once compiled slap it on as a `NativeScript` and add your `nimlib.gdnlib` . I wrote the game of life sample first and then worked on the Dodge the Creeps project, the second time my workflow was _a lot_ faster. Hurray learning and doing!
```

Comment:
~~game of life~~ → Game of Life
~~Hurray learning and doing!~~ → Hurray for learning and doing!